// @magicyan/minecraft/dist/skins/render.mjs
import { skinRenderTypeCrops } from './crops.mjs';
import { RouteBases } from '../constants/api.mjs';
import 'lodash';

function skinRoute(nickOrUUID, type = "default", crop, options) {
  const { camera, lighting, model } = options ?? {};
  const validCrops = skinRenderTypeCrops[type];
  if (crop && !validCrops.includes(crop)) {
    throw new Error(`Invalid crop "${crop}" for render type "${type}". Valid crops are: ${validCrops.join(", ")}`);
  }
  const url = new URL(`${RouteBases.skins}/render/${type}/${nickOrUUID}/${crop ?? validCrops[0]}`);
  if (model)
    loopParams(url, model);
  if (camera)
    loopParams(url, camera);
  if (lighting)
    loopParams(url, lighting);
  return decodeURIComponent(url.toString());
}
async function fetchSkinRender(nickOrUUID, type = "default", crop, options) {
  const response = await fetch(skinRoute(nickOrUUID, type, crop, options));
  if (!response.ok) {
    const { statusText, status } = response;
    return { success: false, error: statusText, code: status };
  }
  const arrayBuffer = await response.arrayBuffer();
  return {
    success: true,
    data: {
      buffer: Buffer.from(arrayBuffer),
      arrayBuffer,
      url: response.url,
      response
    }
  };
}
function loopParams(url, obj) {
  for (const key in obj) {
    const value = obj[key];
    if (typeof value === "object") {
      url.searchParams.set(key, JSON.stringify(value));
      continue;
    }
    if (typeof value !== "undefined") {
      url.searchParams.set(key, `${value}`);
      continue;
    }
  }
}

export { fetchSkinRender, skinRoute };
