// @magicyan/minecraft/dist/index.cjs
'use strict';

const api = require('./constants/api.cjs');
const crops = require('./skins/crops.cjs');
const info = require('./skins/info.cjs');
const render = require('./skins/render.cjs');
const types = require('./skins/types.cjs');
const profile = require('./mojang/profile.cjs');
const session = require('./mojang/session.cjs');
const leaderboard = require('./mush/leaderboard/leaderboard.cjs');
const player = require('./mush/player.cjs');
const base = require('./mush/leaderboard/base.cjs');
const status = require('./server/status.cjs');
const icon = require('./server/icon.cjs');



exports.RouteBases = api.RouteBases;
exports.skinRenderCrops = crops.skinRenderCrops;
exports.skinRenderTypeCrops = crops.skinRenderTypeCrops;
exports.fetchSkinInfo = info.fetchSkinInfo;
exports.fetchSkinRender = render.fetchSkinRender;
exports.skinRoute = render.skinRoute;
exports.skinRenderTypes = types.skinRenderTypes;
exports.fetchMinecraftProfile = profile.fetchMinecraftProfile;
exports.fetchMinecraftProfileSession = session.fetchMinecraftProfileSession;
exports.fetchMushLeaderboard = leaderboard.fetchMushLeaderboard;
exports.fetchMushPlayerInfo = player.fetchMushPlayerInfo;
exports.mushLeaderboardGames = base.mushLeaderboardGames;
exports.fetchServerStatus = status.fetchServerStatus;
exports.isServerOnline = status.isServerOnline;
exports.fetchServerIcon = icon.fetchServerIcon;
