// @magicyan/minecraft/dist/mush/leaderboard/leaderboard.cjs
'use strict';

const result = require('../../helpers/result.cjs');
const format = require('../../helpers/format.cjs');
const api = require('../../constants/api.cjs');
const _ = require('lodash');

function _interopDefaultCompat (e) { return e && typeof e === 'object' && 'default' in e ? e.default : e; }

const ___default = /*#__PURE__*/_interopDefaultCompat(_);

async function fetchMushLeaderboard(game) {
  const url = `${api.RouteBases.mush}/leaderboard/${game}`;
  const response = await fetch(url);
  if (!response.ok)
    return result.notOkResult(response);
  const { records } = await response.json();
  const data = records.map((raw) => ({
    account: format.convertKeysToCamelCase(raw.account),
    avatarURL: raw.avatar_url,
    color: raw.color,
    position: raw.pos,
    game,
    get: implementMethod(game),
    ...___default.omit(raw, ["account", "avatar_url", "color", "pos"])
  }));
  return { success: true, data };
}
function implementMethod(game) {
  const key = game === "skywars" ? "skywars_r1" : game === "hg" ? "hungergames" : game;
  return function getStats(stats) {
    return this[`${key}:${stats}`] ?? -1;
  };
}

exports.fetchMushLeaderboard = fetchMushLeaderboard;
